### ISSUE TITLE
bot error in command

### ISSUE THREAD

**Issue by ismamilahi** [finlaysawyer/discord-uptime#87]
Starting Python Bot
Installing Packages from requirements.txt
Defaulting to user installation because normal site-packages is not writeable
Collecting discord.py==2.0.1
  Downloading discord.py-2.0.1-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 1.2 MB/s eta 0:00:00
Collecting aioping==0.3.1
  Downloading aioping-0.3.1-py3-none-any.whl (11 kB)
Collecting aiohttp~=3.8.3
  Downloading aiohttp-3.8.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 11.2 MB/s eta 0:00:00
Collecting async-timeout
  Downloading async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting aiodns
  Downloading aiodns-3.0.0-py3-none-any.whl (5.0 kB)
Collecting attrs>=17.3.0
  Downloading attrs-22.2.0-py3-none-any.whl (60 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.0/60.0 kB 11.2 MB/s eta 0:00:00
Collecting multidict<7.0,>=4.5
  Downloading multidict-6.0.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (114 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 114.5/114.5 kB 31.3 MB/s eta 0:00:00
Collecting charset-normalizer<3.0,>=2.0
  Downloading charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting aiosignal>=1.1.2
  Downloading aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting yarl<2.0,>=1.0
  Downloading yarl-1.8.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (264 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 264.0/264.0 kB 29.3 MB/s eta 0:00:00
Collecting frozenlist>=1.1.1
  Downloading frozenlist-1.3.3-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (149 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.6/149.6 kB 24.0 MB/s eta 0:00:00
Collecting idna>=2.0
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 8.8 MB/s eta 0:00:00
Collecting pycares>=4.0.0
  Downloading pycares-4.3.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (288 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 288.7/288.7 kB 19.1 MB/s eta 0:00:00
Collecting cffi>=1.5.0
  Downloading cffi-1.15.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 441.8/441.8 kB 18.3 MB/s eta 0:00:00
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 20.9 MB/s eta 0:00:00
Installing collected packages: pycparser, multidict, idna, frozenlist, charset-normalizer, attrs, async-timeout, yarl, cffi, aiosignal, pycares, aiohttp, discord.py, aiodns, aioping
  WARNING: The script normalizer is installed in '/home/container/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed aiodns-3.0.0 aiohttp-3.8.3 aioping-0.3.1 aiosignal-1.3.1 async-timeout-4.0.2 attrs-22.2.0 cffi-1.15.1 charset-normalizer-2.1.1 discord.py-2.0.1 frozenlist-1.3.3 idna-3.4 multidict-6.0.3 pycares-4.3.0 pycparser-2.21 yarl-1.8.2
WARNING | 2022-12-24 01:29:14,753 | discord.client | PyNaCl is not installed, voice will NOT be supported
INFO | 2022-12-24 01:29:14,754 | discord.client | logging in using static token
INFO | 2022-12-24 01:29:16,034 | discord.gateway | Shard ID None has connected to Gateway (Session ID: 29a0e76b4188150617680631eab7401f).
INFO | 2022-12-24 01:29:18,125 | __main__ | Logged in as Royal Node#9477
ERROR | 2022-12-24 01:29:18,132 | discord.client | Ignoring exception in on_ready
Traceback (most recent call last):
  File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 934, in _load_from_module_spec
    spec.loader.exec_module(lib)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/container/cogs/monitor.py", line 8, in <module>
    import aioping
  File "/home/container/.local/lib/python3.10/site-packages/aioping/__init__.py", line 103, in <module>
    proto_icmp = socket.getprotobyname("icmp")
OSError: protocol not found
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/container/.local/lib/python3.10/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/home/container/bot.py", line 45, in on_ready
    await self.load_extension(f"cogs.{filename[:-3]}")
  File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 1012, in load_extension
    await self._load_from_module_spec(spec, name)
  File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 937, in _load_from_module_spec
    raise errors.ExtensionFailed(key, e) from e
discord.ext.commands.errors.ExtensionFailed: Extension 'cogs.monitor' raised an error: OSError: protocol not found

**Comment by ismamilahi** [comment]
how to fix this please tell me

**Comment by Vault108** [comment]
``` py
Traceback (most recent call last):
File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 934, in _load_from_module_spec
spec.loader.exec_module(lib) # type: ignore
File "", line 883, in exec_module
File "", line 241, in _call_with_frames_removed
File "/home/container/cogs/monitor.py", line 8, in
import aioping
File "/home/container/.local/lib/python3.10/site-packages/aioping/init.py", line 103, in
proto_icmp = socket.getprotobyname("icmp")
OSError: protocol not found
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/home/container/.local/lib/python3.10/site-packages/discord/client.py", line 409, in _run_event
await coro(*args, **kwargs)
File "/home/container/bot.py", line 45, in on_ready
await self.load_extension(f"cogs.{filename[:-3]}")
File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 1012, in load_extension
await self._load_from_module_spec(spec, name)
File "/home/container/.local/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 937, in _load_from_module_spec
raise errors.ExtensionFailed(key, e) from e
discord.ext.commands.errors.ExtensionFailed: Extension 'cogs.monitor' raised an error: OSError: protocol not found
```
What does your `server.json` look like?
Just a note that there are only three supported protocols `http`, `tcp` and `ping`

**Comment by alsoGAMER** [comment]
> What operating system are you on?

Afaik, both windows and *nix supports icmp as a protocol, probably this has to do with the containerization software they're using.

**Comment by Vault108** [comment]
> > What operating system are you on?
> 
> Afaik, both windows and *nix supports icmp as a protocol, probably this has to do with the containerization software they're using.


Exactly what I was thinking, if we look at the commands in `ping.py`
https://github.com/finlaysawyer/discord-uptime/blob/439d145d339cbb2dd4cbb93a1b27fb723e4ae5a2/cogs/ping.py#L15-L35
https://github.com/finlaysawyer/discord-uptime/blob/439d145d339cbb2dd4cbb93a1b27fb723e4ae5a2/cogs/ping.py#L36-L57
https://github.com/finlaysawyer/discord-uptime/blob/439d145d339cbb2dd4cbb93a1b27fb723e4ae5a2/cogs/ping.py#L59-L85
We see commands made for ping, http, tcp.  Looking at the code I see no support for ICMP,

**Comment by alsoGAMER** [comment]
> We see commands made for ping, http, tcp. Looking at the code I see no support for ICMP

Yeah because `ping` its actually the name used by the bot code for the icmp protocol.
See https://github.com/stellarbit/aioping/blob/45102cbaf2dad4e73743a343ef94eee27a09d9d5/aioping/__init__.py#L103

tldr: The exception the user is facing is coming from the `socket` library called by `aioping`

**Comment by ismamilahi** [comment]
Brother this is fixed..
sorry i forget to tell you

**Comment by ismamilahi** [comment]
Brother this is fixed..
sorry i forget to tell you
